web-registry/src/components/organisms/BuyCreditsModal/BuyCreditsModal.tsx

```diff
@@ -81,20 +89,62 @@ const BuyCreditsModal: React.FC<BuyCreditsModalProps> = ({
 }) => {
   const styles = useBuyCreditsModalStyles();
   const theme = useTheme();
-  const walletContext = useWallet();
+  const [selectedSellOrder, setSelectedSellOrder] =
+    useState<ISellOrderInfo | null>(null);
 
   const submit = async (values: BuyCreditsValues): Promise<void> => {
-    const recipient = 'regen18hj7m3skrsrr8lfvwqh66r7zruzdvp6ylwxrx4'; // test account
-    const amount = values.creditCount;
-    if (onSubmit) {
-      onSubmit(values);
-    } else if (walletContext.signSend && walletContext.broadcast) {
-      const txBytes = await walletContext.signSend(amount, recipient);
-      onTxQueued && onTxQueued(txBytes);
-      onClose();
+    const { country, postalCode, stateProvince, retirementAction } = values;
+
+    if (onSubmit && selectedSellOrder) {
+      const fullValues: BuyCreditsValues = {
+        ...values,
+        price: parseInt(selectedSellOrder.askAmount),
+        batchDenom: selectedSellOrder.batchDenom,
+        sellOrderId: selectedSellOrder.id,
+        country: retirementAction === 'autoretire' ? country : '',
+        postalCode: retirementAction === 'autoretire' ? postalCode : '',
+        stateProvince: retirementAction === 'autoretire' ? stateProvince : '',
+      };
+      onSubmit(fullValues);
     }
   };
 
+  const getOptions = (): Option[] => {
+    if (project?.sellOrders?.length) {
+      const sellOrderOptions = project?.sellOrders.map(sellOrder => {
+        return {
+          label: getSellOrderLabel(sellOrder),
+          value: sellOrder.id,
+        };
+      });
+      return [{ label: 'Choose a sell order', value: '' }, ...sellOrderOptions];
+    }
+
+    return [];
+  };
+
+  const SetSelectedSellOrder = (): JSX.Element => {
+    const { values, setFieldValue } = useFormikContext<BuyCreditsValues>();
+    if (selectedSellOrder?.id !== values.sellOrderId) {
+      const _selectedSellOrder = project?.sellOrders?.find(
+        sellOrder => sellOrder.id === values?.sellOrderId,
+      );
+      if (_selectedSellOrder) {
+        setSelectedSellOrder(_selectedSellOrder);
+        if (
+          _selectedSellOrder.disableAutoRetire &&
+          values.retirementAction !== 'manual'
+        ) {
+          setFieldValue('retirementAction', 'manual');
+        }
+      } else {
+        setSelectedSellOrder(null);
+      }
```

Comment:
We could simplify the logic a bit:

```suggestion
      setSelectedSellOrder(_selectedSellOrder ?? null);
      
      if (
        _selectedSellOrder.disableAutoRetire &&
        values.retirementAction !== 'manual'
      ) {
        setFieldValue('retirementAction', 'manual');
      }
```

Reply:
thx